addons/mail/static/src/models/user_setting/user_setting.js

```diff
@@ -179,8 +178,8 @@ registerModel({
          * @param {any} key
          */
         _debounce(f, delay, key) {
-            this._timeoutIds[key] && browser.clearTimeout(this._timeoutIds[key]);
-            this._timeoutIds[key] = browser.setTimeout(() => {
+            this.timeoutIds[key] && browser.clearTimeout(this.timeoutIds[key]);
+            this.timeoutIds[key] = browser.setTimeout(() => {
```

Comment:
It probably doesn't make a difference here, but writing on a field without `update()` shouldn't be done.

Reply:
Agree, I just turn into fields in a straight-forward way.
Could introduce a `Dict` model, but I felt it's out of scope of PR and good improvement at later date,

Reply:
will do that then

Reply:
Just did that for all `Object` attributes introduced in this PR (and the `Set` has been turned into `Object` attribute).
Not super pretty with spread object for update (and deletion of item in 3 statements), but changes on these fields are now observed.